src/stm32f103/target_stm32f103.c

```diff
@@ -161,6 +161,10 @@ bool target_get_force_bootloader(void) {
     backup_write(BKP0, 0);
 
 #if HAVE_BUTTON
+    /* Wait sometime in case the button has some debounce capacitor */
+    for (int i = 0; i < BUTTON_SAMPLE_DELAY_CYCLES; i++) {
```

Comment:
Sorry, I just remembered that on my "official build laptop" ;p that I don't have with me at the moment, the super old stable version 4.8 of gcc-arm defaults to C89 and doesn't allow C++ style `for (int i=0; ...)`.

Can you change it to the lame but ancient-compiler-friendly:
```cpp
int i;
for (i = 0; i < BUTTON_SAMPLE_DELAY_CYCLES; i++) {
    ....
}
```

Normally, I would save these kind of nitpicky compiler compatibility things until I actually have my ancient compiler in front of me, but I won't be able to test anything for a couple of days and I see that you're working off of your master branch, so I figured it would be better to give you feedback sooner than later.

Reply:
Done

Reply:
Upon reviewing my Makefile, apparently I got fed up with C89 for this project and set the standard to `gnu11`, so I guess it wouldn't have mattered either way.